crates/runtime/src/dataconnector/mod.rs

```diff
@@ -431,6 +452,12 @@ pub trait DataConnectorFactory: Send + Sync {
     ///
     /// Any parameter provided by a user that isn't in this list will be filtered out and a warning logged.
     fn parameters(&self) -> &'static [ParameterSpec];
+
+    /// Returns a list of keywords that are protected by the data connector.
+    /// Used to ensure that any table name isn't a protected keyword.
+    fn protected_keywords(&self) -> &'static [&'static str] {
```

Comment:
```suggestion
    fn reserved_keywords(&self) -> &'static [&'static str] {
```

I think `reserved` is the more idiomatic phrase?